src/main/java/com/kodilla/ecommercee/controller/ProductController.java

```diff
@@ -1,39 +1,68 @@
+import com.kodilla.ecommercee.service.ProductDbService;
+import lombok.RequiredArgsConstructor;
 import org.springframework.http.MediaType;
 import org.springframework.web.bind.annotation.*;
 
-import java.math.BigDecimal;
-import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
 
+@RequiredArgsConstructor
 @RestController
 @RequestMapping("/v1/product")
-public class
-ProductController {
-    @RequestMapping(method = RequestMethod.GET, value = "getProducts")
+public class ProductController {
+
+    private final ProductMapper productMapper;
+    private final ProductDbService productDbService;
+
+
+    @GetMapping(value = "getProducts")
     public List<ProductDto> getProducts() {
-        return new ArrayList<>();
+        return productMapper.mapToProductDtoList(productDbService.getAllProducts());
     }
 
     @GetMapping(value = "getProduct")
-    public ProductDto getProduct(@RequestParam Long productId) {
-        return new ProductDto(1L, "name", "description", new BigDecimal("100"), "1");
+    public ProductDto getProduct(@RequestParam Long productId) throws ProductNotFoundException {
+        return productMapper.mapToProductDto(
+                productDbService.getProductById(productId).orElseThrow(ProductNotFoundException::new));
     }
 
     @PostMapping(value = "createProduct", consumes = MediaType.APPLICATION_JSON_VALUE)
-    public void createProduct(@RequestBody ProductDto productDto) {
+    public ProductDto createProduct(@RequestBody ProductDto productDto) throws ProductsGroupNotFoundException {
+            Product product = productMapper.mapToProduct(productDto);
+            Product createdProduct = productDbService.saveProduct(new Product(
+                    product.getName(),
+                    product.getDescription(),
+                    product.getPrice(),
+                    product.getProductsGroup()
+            ));
+            return productMapper.mapToProductDto(createdProduct);
     }
 
     @PutMapping(value = "updateProduct", consumes = MediaType.APPLICATION_JSON_VALUE)
-    public ProductDto updateProduct(@RequestBody ProductDto productDto) {
-        return new ProductDto(1L, "name", "description", new BigDecimal("100"), "1");
+    public ProductDto updateProduct(@RequestBody ProductDto productDto) throws Exception {
+        Product product = productMapper.mapToProduct(productDto);
+        Optional<Product> productFromDb = productDbService.getProductById(productDto.getId());
+        if (productFromDb.isPresent()) {
+            Product productToUpdate = productFromDb.get();
+            productToUpdate.setName(product.getName());
+            productToUpdate.setDescription(product.getDescription());
+            productToUpdate.setPrice(product.getPrice());
+            productToUpdate.setProductsGroup(product.getProductsGroup());
+            return productMapper.mapToProductDto(productDbService.saveProduct(productToUpdate));
```

Comment:
Wydaje mi się, że możesz tu zrobić krócej, czyli jeśli sprawdziłeś sobie, że produkt o danym id istnieje, to w returnie zamiast przekazywać productToUpdate możesz już przekazać zmienną product. Te linijki 53-57 wydają się zbędne. Przetestowałam krótszą wersję i wygląda, że dobrze działa (podobnie robiliśmy w TaskControllerze w zadaniu z kursu)

Reply:
sto procent racja ;) już zmieniłem